MetaMask/metamask-mobile
# chore: Fix handling of generated ppom files

## **Description**

- Add generated `app/lib/ppom/ppom.html.js` to `.gitignore` and remove file from source control
- Infer webpack build mode for ppom subproject from `NODE_ENV` env var
  - This means that dev builds will not be minified (like the production build being ignored here is)
- Add `clean:ppom` package script which cleans generated ppom files
  - Run `clean:ppom` as part of `clean` package script
- ci: Run `yarn setup` (includes ppom-build step in addition to the steps in `setup:node`) instead of `yarn setup:node` before GitHub Actions unit tests 
  
## **Related issues**

- #9459
- https://github.com/MetaMask/metamask-mobile/pull/9454#discussion_r1584720941

## **Manual testing steps**
- Run `rm app/lib/ppom/ppom.html.js && NODE_ENV=production yarn setup`
  * Verify that `app/lib/ppom/ppom.html.js` is identical to previous version, e.g. by comparing output of `sha256sum app/lib/ppom/ppom.html.js`
- Run `rm app/lib/ppom/ppom.html.js && NODE_ENV=development yarn setup`
  * Verify that `app/lib/ppom/ppom.html.js` is not minified

## **Screenshots/Recordings**

### **Before**

### **After**

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
